### PR TITLE
Gitlab CI improvements

### PR DESCRIPTION
**Pull request by wz1000**
This patch improves the release scripts in a number of ways and unblocks
releases on darwin due to LLVM mismatches.

- Add ubuntu builds to gitlab CI
- Run pipelines from web interface
- Use nix for darwin builds instead of homebrew
- Drop 9.2.3 and add 9.4.3
- Build in parallel

<a href="https://gitpod.io/#https://github.com/haskell/haskell-language-server/pull/3324"></a>

### CODE REVIEW

File: .gitlab/darwin/toolchain.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{ system }:
```
Comment (michaelpj) [inline .gitlab/darwin/toolchain.nix:1]: what is this for? what does it do? etc. etc. documentation please!

Comment (wz1000) [reply .gitlab/darwin/toolchain.nix:1]: This is taken almost verbatim from the GHC nix setup script. I can add a note pointing to that, but beyond that I have very little experience with nix, so all of the nix stuff is just cargo-culted from GHC itself.

Comment (michaelpj) [reply .gitlab/darwin/toolchain.nix:1]: "I cargo-culted this from GHC, here's a link to the upstream version" is _incredibly_ useful information for someone reading this!

**Comment by hasufell** [comment]
Looks ok to me... but I won't be touching anything nix related. So hope you're up for maintaining that.

**Comment by pepeiborra** [comment]
@wz1000 do you still need a stamp on this?

**Review by michaelpj — APPROVED** [review]

**Pull request merged**